dnf/base.py

```diff
@@ -106,24 +106,52 @@ def _add_tempfiles(self, files):
         else:
             self._tempfiles.update(files)
 
-    def _add_repo_to_sack(self, repo):
-        repo.load()
-        hrepo = repo._hawkey_repo
-        hrepo.repomd_fn = repo._repomd_fn
-        hrepo.primary_fn = repo._primary_fn
-        hrepo.filelists_fn = repo._filelists_fn
-        hrepo.cost = repo.cost
-        if repo._presto_fn:
-            hrepo.presto_fn = repo._presto_fn
-        else:
-            logger.debug("not found deltainfo for: %s", repo.name)
-        if repo._updateinfo_fn:
-            hrepo.updateinfo_fn = repo._updateinfo_fn
-        else:
-            logger.debug("not found updateinfo for: %s", repo.name)
-        self._sack.load_repo(hrepo, build_cache=True, load_filelists=True,
-                             load_presto=repo.deltarpm,
-                             load_updateinfo=True)
+    def _add_repos_to_sack(self, repos_iterable):
+        """
+        @param repos_iterable: Sequence of repos in iterable form
```

Comment:
@ignatenkobrain But it is used in dnf. Or be verbose that everyone will understand what you mean!!!